src/librustc_mir/interpret/eval_context.rs

```diff
@@ -492,23 +492,6 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tc
         let frame = self.stack.pop().expect(
             "tried to pop a stack frame, but there were none",
         );
-        match frame.return_to_block {
-            StackPopCleanup::Goto(block) => {
-                self.goto_block(block)?;
-            }
-            StackPopCleanup::None { cleanup } => {
-                if !cleanup {
-                    // Leak the locals. Also skip validation, this is only used by
```

Comment:
Now we do the validation twice for the result of constants. Even if one of the validations is weaker, we might still get hit with perf regressions

Reply:
CTFE doesn't do validation during execution, it opts out of that.

Reply:
But anyway this got resolved by the 2nd commit.